text/0000-portals.md

```diff
@@ -0,0 +1,110 @@
+
+## Detailed design
+
+(not an actual detailed design, just adding some possible requirements)
+
+- If a custom element has `lwc:portal` attribute on it, and its value is an existing global id, instead of being inserted at its natural position in the template, the element would be inserted in the element with the defined global id.
+- When an element's destination portal doesn't exist, but later is inserted in the DOM, any elements that have already rendered stay in their original location, as if the portal doesn't exist.
+- When the host element is disconnected, the portalled element should also be disconnected. 
+- The portalled component should still be `@Context` aware 
+- The portalled element should only be allowd at the top root of the template, this way any CSS issues are avoided
+- Any CSS defined should apply to the content of the portalled element
+- An element can't portal to itself (`lwc:portal` matches the `id`)
+
+
+## Drawbacks
+
+- new API
+- additional complexity in LWC
+- more complexity around event propagation
+- issues with CSS unless the portalled element is forced to be at the top root of the template
+
+## Security
+
+- The portalled element content should not be accessible by anyone but the originally owning host element.
+
+## Alternatives
+
+In case of dialogs, some other strategies have been considered:
+1. Event based creation of a dialog with a dialog manager defined at some parent node. 
+Given that a modal could have up to 3 slots (header, default, footer), the ergonomics of usage become very cumbersome.
+Moreover event-based modal creation is inferior to `<dialog>` like experience.
+2. Exposing top layer by the browsers. This would allow one to effectively declare a new top layer and override any 
+existing stacking context. It's unlikely that proposal would support any kind of scoping, so this wouldn't work either.
+
+## Adoption strategy
+
+It's not a breaking change, but it does introduce new API, adoption would initially be from component libraries with
+support for modals and popups.
+
+## How we teach this
+
+This would allow implementation of components that can ignore the current stacking context. Teaching should be around what components can benefit from this.
+
+## Unresolved questions
+
+1. How would `querySelector` work ?
+2. Is the definiton of the portal immutable (can you change its name, what if it gets removed?)
+3. Can the destional portal be changed ?
+4. What if the portal doesn't exist
+5. What if there are more than 1 portal definition with the same name
+6. Should there be implicit/reserved portals ? (eg. `body` which always implies the `<body>` element of the document) 
```

Comment:
Seeing as body is the most likely default it makes sense that `lwc:portal` with no value would be set to `<body>`.